examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -546,66 +549,85 @@ func (g *genDeepCopy) doSlice(t *types.Type, sw *generator.SnippetWriter) {
 }
 
 func (g *genDeepCopy) doStruct(t *types.Type, sw *generator.SnippetWriter) {
-	if len(t.Members) == 0 {
-		// at least do something with in/out to avoid "declared and not used" errors
-		sw.Do("_ = in\n_ = out\n", nil)
-		return
-	}
-
 	if hasDeepCopyMethod(t) {
 		sw.Do("*out = in.DeepCopy()\n", nil)
 		return
 	}
 
+	// Simple copy covers a lot of cases.
+	sw.Do("*out = *in\n", nil)
+
+	// Now fix-up fields as needed.
 	for _, m := range t.Members {
 		t := m.Type
+		hasMethod := false
+		if hasDeepCopyMethod(t) {
+			hasMethod = true
+		}
 		if t.Kind == types.Alias {
 			copied := *t.Underlying
 			copied.Name = t.Name
 			t = &copied
 		}
 		args := generator.Args{
 			"type": t,
+			"kind": t.Kind,
 			"name": m.Name,
 		}
 		switch t.Kind {
-		case types.Builtin:
-			sw.Do("out.$.name$ = in.$.name$\n", args)
+		case types.Builtin, types.Array:
+			if hasMethod {
+				sw.Do("out.$.name$ = in.$.name$.DeepCopy()\n", args)
+			}
 		case types.Map, types.Slice, types.Pointer:
-			sw.Do("if in.$.name$ != nil {\n", args)
-			sw.Do("in, out := &in.$.name$, &out.$.name$\n", args)
-			g.generateFor(t, sw)
-			sw.Do("} else {\n", nil)
-			sw.Do("out.$.name$ = nil\n", args)
-			sw.Do("}\n", nil)
+			if hasMethod {
+				sw.Do("if in.$.name$ != nil {\n", args)
+				sw.Do("    out.$.name$ = in.$.name$.DeepCopy()\n", args)
```

Comment:
We explicitly stopped adding spaces at the beginning, because we are calling gofmt at the end anyway. I think you should also remove them here and below - this is painful to maintain them.